.github/workflows/smoke-test-release.yml

```diff
@@ -1,185 +1,690 @@
+              run: npm install axios
+
+            - name: Get version numbers
+              id: get-versions
+              uses: actions/github-script@v6
+              with:
+                  script: |
+                      const { getPreviousTwoVersions } = require('./script/wordpress');
+                      const versions = await getPreviousTwoVersions();
+                      console.log(versions);
+                      core.setOutput('versions', versions);
+
+    test-wp-versions:
+        name: Test against ${{ matrix.version.description }} (${{ matrix.version.number }})
+        runs-on: ubuntu-20.04
+        needs: [get-wp-versions]
+        strategy:
+            matrix: ${{ fromJSON(needs.get-wp-versions.outputs.matrix) }}
+        env:
+            API_ALLURE_REPORT_DIR: ${{ github.workspace }}/plugins/woocommerce/tests/e2e-pw/test-results/api/allure-report
+            API_ALLURE_RESULTS_DIR: ${{ github.workspace }}/plugins/woocommerce/tests/e2e-pw/test-results/api/allure-results
+            API_WP_LATEST_X_ARTIFACT: api-${{ matrix.version.env_description }}--run-${{ github.run_number }}
+            E2E_ALLURE_REPORT_DIR: ${{ github.workspace }}/plugins/woocommerce/tests/e2e-pw/test-results/e2e/allure-report
+            E2E_ALLURE_RESULTS_DIR: ${{ github.workspace }}/plugins/woocommerce/tests/e2e-pw/test-results/e2e/allure-results
+            E2E_WP_LATEST_X_ARTIFACT: e2e-${{ matrix.version.env_description }}--run-${{ github.run_number }}
+        permissions:
+            contents: read
+        steps:
+            - name: Checkout WooCommerce repo
+              uses: actions/checkout@v3
 
             - name: Setup WooCommerce Monorepo
               uses: ./.github/actions/setup-woocommerce-monorepo
+
+            - name: Launch WP Env
+              working-directory: plugins/woocommerce
+              run: pnpm run env:test
+
+            - name: Download release zip
+              env:
+                  GITHUB_TOKEN: ${{ github.token }}
+              run: gh release download ${{ github.event.inputs.tag }} --dir tmp
+
+            - name: Replace `plugins/woocommerce` with unzipped woocommerce release build
+              run: unzip -d plugins -o tmp/woocommerce.zip
+
+            - name: Downgrade WordPress version to ${{ matrix.version.number }}
+              working-directory: plugins/woocommerce
+              run: |
+                  pnpm exec wp-env run tests-cli "wp core update --version=${{ matrix.version.number }} --force"
+                  pnpm exec wp-env run tests-cli "wp core update-db"
+
+            - name: Verify environment details
+              working-directory: plugins/woocommerce
+              run: |
+                  pnpm exec wp-env run tests-cli "wp core version"
+                  pnpm exec wp-env run tests-cli "wp plugin list"
+                  pnpm exec wp-env run tests-cli "wp theme list"
+                  pnpm exec wp-env run tests-cli "wp user list"
```

Comment:
As we've encountered certain WP ENV issues, I think it's good to print out the site setup before running the tests.